infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/jwtproxy/JwtProxyProvisioner.java

```diff
@@ -199,7 +198,20 @@ public ServicePort expose(
             .withNewTargetPort(listenPort)
             .build();
 
-    k8sEnv.getServices().get(getServiceName()).getSpec().getPorts().add(exposedPort);
+    k8sEnv.getServices().get(serviceName).getSpec().getPorts().add(exposedPort);
+
+    proxyConfigBuilder.addVerifierProxy(
+        listenPort,
+        "http://" + backendServiceName + ":" + backendServicePort.getTargetPort().getIntVal(),
+        excludes,
+        cookiesAuthEnabled,
+        identity.getWorkspaceId(),
```

Comment:
It would be useful to mention in PR description that cookies mechanism is a little patched: now path rules are changed from `{che-host}://theia-blahblah-432.192.168.99.100.nip.io` to `{che-host}://workspace123acsd/theia-blahblah-432.192.168.99.100.nip.io` and one cookie is used for all workspace servers

Reply:
That is not exactly how it works.

The cookie is required only for JWT-fronted endpoints. These endpoints were modified to deploy to `{che-host}/{workspace-id}/{service-name}/{service-port}` (originally it deployed to `{che-host}/{service-name}/{service-port}`).

The auth cookie originally bound to the path `/` which means that it was valid for the whole domain it deployed to. This worked in multi-host mode because each service was its own domain, but fails miserably in single-host mode, because this essentially means we have a single auth cookie for all workspaces, which is not correct, because the auth mechanism works per-workspace.

Now the auth cookie binds to `/{workspace-id}` which means that each `{che-host}/{workspace-id}` needs its own cookie - i.e. in the past we had a single auth cookie for the whole che server, and all workspaces running underneath, but now each workspace can have its own auth cookie.